x-pack/legacy/plugins/maps/public/layers/layer.js

```diff
@@ -359,5 +362,9 @@ export class AbstractLayer {
     return [];
   }
 
+  syncVisibilityWithMb(mbMap, mbLayerId) {
```

Comment:
Creating this function is a nice consolidation of duplicate logic in multiple places. This is much better